save-orchestrator/src/main/kotlin/com/saveourtool/save/orchestrator/controller/AgentsController.kt

```diff
@@ -228,9 +230,16 @@ class AgentsController(
         .map {
             log.info { "Downloaded all tests for execution $id to $targetDirectory" }
         }
-        .defaultIfEmpty(log.warn {
-            "Not found any tests for execution ${requiredId()}"
-        })
+        .lazyDefaultIfEmpty {
+            log.warn {
+                "Not found any tests for execution ${requiredId()}"
+            }
+        }
+
+    private fun Flux<DataBuffer>.writeTo(targetFile: Path): Mono<Path> =
+            DataBufferUtils.write(this, targetFile.outputStream())
+                .map { DataBufferUtils.release(it) }
+                .then(Mono.just(targetFile))
```

Comment:
Is there any advantage on wrapping the value instead of calling `thenReturn`?

Reply:
funny, but Flux doesn't have `thenReturn` (I wanted to use it)